crates/driver/src/infra/pod/api.rs

```diff
@@ -0,0 +1,257 @@
+}
+
+fn make_signer(
+    eth_account: ethcontract::Account,
+) -> Result<Box<dyn TxSigner<Signature> + Send + Sync>, Error> {
+    match eth_account {
+        Account::Offline(pk, chain_id) => {
+            let bytes = pk.as_ref();
+            let key = k256::ecdsa::SigningKey::from_slice(bytes)
+                .map_err(|e| Error::FailedToConnect(format!("invalid signing key: {e}")))?;
+
+            let signer = LocalSigner::from(key).with_chain_id(chain_id);
+            Ok(Box::new(signer))
+        }
+        Kms(kms_account, _) => {
+            let signer = tokio::runtime::Handle::current().block_on(async {
+                AwsSigner::new(
+                    kms_account.client().clone(),
+                    kms_account.key_id().to_string(),
+                    None,
+                )
+                .await
+                .map_err(|e| Error::FailedToConnect(format!("failed to create AwsSigner: {e}")))
+            })?;
+            Ok(Box::new(signer))
+        }
+        other => Err(Error::FailedToConnect(format!(
+            "unsupported eth_account variant: {other:?}"
+        ))),
+    }
+}
+
+pub struct Pod {
+    endpoint: Url,
+    pub explorer: Url,
+    contract_address: alloy_primitives::Address,
+    pod_provider: PodProvider,
+}
+
+impl Pod {
+    // Initialize a new Pod instance with the given account.
+    pub async fn new(
+        endpoint: Url,
+        explorer: Url,
+        contract_address: Address,
+    ) -> Result<Self, Error> {
+        let pod_provider = PodProviderBuilder::with_recommended_settings()
+            .on_url(endpoint.clone())
+            .await
+            .expect("pod provider should build");
+        Ok(Self {
+            endpoint,
+            explorer,
+            contract_address: alloy_primitives::Address::from_slice(contract_address.as_bytes()),
+            pod_provider,
+        })
+    }
+
+    pub async fn bid(
+        &self,
+        account: Account,
+        id: i64,
+        deadline: u64,
+        value: U256,
+        data: &[u8],
+    ) -> Result<web3::types::H256, Error> {
+        let signer = make_signer(account)?;
+        let wallet = EthereumWallet::from(signer);
+
+        // We want to instantiate a new PodProvider with the bider's wallet assigned
+        let local_pod_provider = PodProviderBuilder::with_recommended_settings()
+            .wallet(wallet)
+            .on_url(self.endpoint.clone())
+            .await
+            .map_err(|e| Error::FailedToConnect(format!("failed to build provider: {e}")))?;
+
+        let auction_client = AuctionClient::new(local_pod_provider, self.contract_address);
+
+        let pod_auction_id =
+            pod_sdk::U256::from(u64::try_from(id).map_err(|_| Error::InvalidAuctionId(id))?);
+        let pod_auction_value = to_pod_u256(value);
+
+        let receipt = auction_client
+            .submit_bid(
+                pod_auction_id,
+                UNIX_EPOCH + Duration::from_micros(deadline),
```

Comment:
The solver already gets a timestamp in the `/solve` request. Wouldn't it make more sense to convert 1 timestamp type to another here instead of converting to `u64` in the middle?